crates/revm/src/optimism.rs

```diff
@@ -21,6 +21,7 @@ const NON_ZERO_BYTE_COST: u64 = 16;
 /// uint64 _sequenceNumber, bytes32 _batcherHash, uint256 _l1FeeOverhead, uint256 _l1FeeScalar)
 ///
 /// For now, we only care about the fields necessary for L1 cost calculation.
+#[derive(Clone, Copy)]
```

Comment:
why is this needed? where are we cloning it? Not a big deal anyway, but just want to make sure it's needed

Reply:
Now that `l1_block_info` is an Optional type, the inner `L1BlockInfo` must be either moved or copied when it's accessed in each loop iteration. If we don't derive clone/copy, then the compiler rejects the use of a moved value.

Reply:
Awesome!